src/vs/platform/webview/electron-main/webviewMainService.ts

```diff
@@ -57,4 +62,43 @@ export class WebviewMainService extends Disposable implements IWebviewManagerSer
 			contents.setIgnoreMenuShortcuts(enabled);
 		}
 	}
+
+	public async findInFrame(windowId: WebviewWindowId, frameName: string, text: string, options: { findNext?: boolean, forward?: boolean }): Promise<void> {
+		const frame = this.getFrameByName(windowId, frameName);
+		frame.findInFrame(text, {
+			findNext: options.findNext,
+			forward: options.forward,
+		});
+		if (!this.frameHandlers.has(frameName)) {
+			const event = frame.on('found-in-frame', (_, result) => {
+				if (result.finalUpdate) {
+					this._onFoundInFrame.fire(result);
+				}
+			});
+			this.frameHandlers.set(frameName, event);
```

Comment:
I am missing some code that clears the frame handler when done, does this end up piling events up, then it sounds like a potential memory leak?

Reply:
Changed around the code to not require `frameHandlers` variable.